include/storage/shared_memory.hpp

```diff
@@ -322,47 +322,38 @@ class SharedMemory
 };
 #endif
 
-template <class LockFileT = OSRMLockFile> class SharedMemoryFactory_tmpl
+template <typename IdentifierT, typename LockFileT = OSRMLockFile>
+static SharedMemory *makeSharedMemory(const IdentifierT &id,
```

Comment:
Remove static, duplicates code across translation units. Does not violate the ODR because it's a non-fully specialized template -> inline by default, but you could also make it explicit by replacing static with inline.